tests/by-util/test_csplit.rs

```diff
@@ -1342,3 +1342,33 @@ fn test_line_num_range_with_up_to_match3() {
     assert_eq!(at.read("xx01"), "");
     assert_eq!(at.read("xx02"), generate(10, 51));
 }
+
+#[test]
+fn precision_format() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["numbers50.txt", "10", "--suffix-format", "%#6.3x"])
+        .succeeds()
+        .stdout_only("18\n123\n");
+
+    let count = glob(&at.plus_as_string("xx*"))
+        .expect("there should be splits created")
+        .count();
+    assert_eq!(count, 2);
+    assert_eq!(at.read("xx   000"), generate(1, 10));
+    assert_eq!(at.read("xx 0x001"), generate(10, 51));
+}
+
+#[test]
+fn precision_format2() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["numbers50.txt", "10", "--suffix-format", "%0#6.3x"])
+        .succeeds()
+        .stdout_only("18\n123\n");
+
+    let count = glob(&at.plus_as_string("xx*"))
+        .expect("there should be splits created")
+        .count();
+    assert_eq!(count, 2);
+    assert_eq!(at.read("xx   000"), generate(1, 10));
+    assert_eq!(at.read("xx 0x001"), generate(10, 51));
+}
```

Comment:
At first glance those test functions  look identical and it's difficult to see what the difference is: a single `0`. So my suggestion is to merge the functions and loop over the suffix formats.

Reply:
Great suggestion! I'll get right on it